pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -91,7 +93,17 @@ func GetPublicZone(name string) (*route53.HostedZone, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "getting AWS session")
 	}
-	client := route53.New(session)
+	// TODO Remove the hard code api endpoint after it's GA in AWS China
+	// If it's AWS China region, we have to set region to "cn-northwest-1"
+	// as route53 is hosted in that region.
+	awsConfig := aws.NewConfig()
+	switch region {
+	case endpoints.CnNorth1RegionID, endpoints.CnNorthwest1RegionID:
+		awsConfig.WithRegion(endpoints.CnNorthwest1RegionID)
+		awsConfig.WithEndpoint("https://route53.amazonaws.com.cn")
+	}
+	client := route53.New(session, awsConfig)
+	logrus.Debugf("listing AWS hosted zones")
```

Comment:
i would rather see this function receive the route53 client, instead of it building one internally here.

Reply:
Let's keep this for now, finally we will remove this logic after route53 in China is GA.